AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/TestSupport/AsyncExpectation.swift

```diff
@@ -1,196 +0,0 @@
-//
-// Copyright Amazon.com Inc. or its affiliates.
-// All Rights Reserved.
-//
-// SPDX-License-Identifier: Apache-2.0
-//
-
-import Foundation
-import XCTest
-import Amplify
-
-public actor AsyncExpectation {
```

Comment:
DataStoreHostApp/APIHostApp, which has integration test targets, pulls in package.swift which has the following products available:
```
products: [
        .library(
            name: "Amplify",
            targets: ["Amplify"]
        ),
        .library(
            name: "AWSAPIPlugin",
            targets: ["AWSAPIPlugin"]
        ),
        .library(
            name: "AWSCognitoAuthPlugin",
            targets: ["AWSCognitoAuthPlugin"]
        ),
        .library(
            name: "AWSDataStorePlugin",
            targets: ["AWSDataStorePlugin"]
        ),
        .library(
            name: "AWSS3StoragePlugin",
            targets: ["AWSS3StoragePlugin"]
        ),
        .library(
            name: "AWSLocationGeoPlugin",
            targets: ["AWSLocationGeoPlugin"]
        ),
        .library(
            name: "AWSPinpointAnalyticsPlugin",
            targets: ["AWSPinpointAnalyticsPlugin"]
        )
    ],
```

We do not publiclly expose AmplifyTestCommon, which is the reason why this file was duplicated. By removing it here, the integration test targets are broken since they still don't have access to AmplifyTestCommon, and if i understand correctly, this PR only moves the AsyncExpectation and related under AmplifyAsyncTesting, which is still under AmplifyTestCommon

Reply:
All integration tests are opened via their host app's xcodeproj file, for example DataStoreHostApp and APIHostApp. switching to an test target, uses the host app as the host application. Building fails because the file is removed:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/1365977/187758879-8e12f56c-7a94-4baa-a663-aedb67a08af4.png">

Reply:
Yeah you're right, we need to add `AmplifyAsyncTesting`  here if we want to use them in the integration tests host apps.

Reply:
integration testing is done via the host apps, ie. DataStoreHostApp and APIHostApp. Each test target runs on the host app, which has package.swift added to the project, the hostapp adds the products needed like the plugin targets. the hostapp and test targets do not have access to AsyncExpectation because it doesn't have access to AmplifyTestCommon/AmplifyAsyncTesting

<img width="365" alt="image" src="https://user-images.githubusercontent.com/1365977/187759681-157fbda1-176c-4e0a-8dbe-5378d738f215.png">